### ISSUE TITLE
Samplesheets: Initial UCSC Genome Browser integration

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#238]
This is the task for initial UCSC Genome Browser integration into the UI.

## Use Case

- UCSC integration requires **track hub** collection with files accessible with a URL
- We need to look for a `TrackHubs` collection under each assay, which stores 0-N track hubs created by users
    * Currently assay specific, may be expanded to e.g. study specific in the future
    * Track hubs are created by uploading a collection and related files under `TrackHubs` via landing zones
- We need to enable providing ticket-based Davrods access to user-created track hub collections in iRODS
- There needs to be management of these tickets in the samplesheets app UI
- iRODS links to these track hubs need to be displayed in the UI along with links to the ticket access URLs

## Requirements

- Add new Django settings
    * `IRODS_WEBDAV_URL_ANON`: URL to webdav with anon/ticket access (default=`IRODS_WEBDAV_URL`)
    * `IRODS_WEBDAV_USER_ANON`: Name of user for anon/ticket access (default=`ticket`)
- Add `IrodsAccessTicket` model for tickets in `samplesheets.models` (extendable for other types of collections in the future)
    * `project`: foreign ref to `Project`
    * `study`: foreign ref to `Study`
    * `assay`: foreign reff to `Assay`
        + NOTE: Can be null even though we won't allow it in the UI yet
    * `ticket`: Char field
    * `path`: Char field to irods path
    * `label`: Char field for a free text label (optional? default generated from e.g. assay/user name?)
    * `user`: Foreign ref to `User` who created this
    * `date_created`
    * `date_expires`
- Add UI for managing tickets
    * Do this as a Django view (similar to e.g. sheet versions)
    * Project specific
    * Link from *Sheet Operations*
    * Operations
        + List
        + Create
            + This needs to create a ticket in iRODS using `irodsbackend` (obviously)
            + Only allow creation for existing collections which have been created directly under an assay's `TrackHubs` collection (by landing zone import)
        + Update (rename, change expiry date?)
        + Delete
    * Perms
        + Users with sheet read access can see all tickets but not edit
        + Users with sheet edit access can edit/delete their own tickets
        + Superuser/owner/delegate can edit all
    * Fields visible to user for creation
        + `path`
            + We can derive get the `study` and `assay` from this
            + Need to query for the project's assays and related `TrackHubs/*` collections using `irodsbackends` before displaying the form
            + Choices should be existing collections **directly** below each `TrackHubs` collection
            + Label for selection could be something like `${assay display name} / ${collection name}`
        + `label`
        + `date_expires`
            + Calendar input + additional field for "no expiry?"
- Add UI linking for track hub collections
    * In each assay, **for each track hub** under `TrackHubs`, display an additional "shortcut box" in *assay shortcuts*
        + Similar to existing shortcuts added by assay sub-apps
        + Extend the current implementation at:
            + `samplesheets.plugins.update_cache()`
            + `samplesheets.views_ajax.SampleSheetStudyTablesAjaxView`
            + `AssayShortcutCard` and `IrodsButtons` in the vue app
            + (NOTE: Study shortcuts are a differnet thing and not related, at least for now)
        + **TBD:** Label shown in each shortcut box? (e.g. label if present in model, otherwise name of collection under `TrackHubs`?)
            + The label should be something short so the shortcut box won't get huge (or, add truncation and tooltip for full name?)
        + If there is nothing under `TrackHubs` or the collection doesn't even exist, don't show anything
    * Add an additional anon/ticket URL button to the links
        + URL format: `https://${IRODS_WEBDAV_USER_ANON}:${TICKET}/${IRODS_WEBDAV_USER_ANON}/${PATH}`
        + If multiple tickets exist, link to the newest one which hasn't expired
            + Later, once we implement something like IP restrictions per ticket, we can expand this to open a modal for multiple tickets
- Technical implementation of the UI addition
    * Add functionality to display optional extra buttons in `IrodsButtons`
        + Add prop `extraButtons`
        + Prop should be in format: `[{icon: 'fa-something', title: 'Title shown in tooltip', url: 'url', enabled: boolean}]`
        + Add these to the beginining of the button list
    * Provide these to `IrodsButtons` in `AssayShortcutCard`
        + The URL and track hub label are provided by `SampleSheetStudyTablesAjaxView` on server side (see above), other parameters can be simply set in `AssayShortcutCard` because they are always the same in this case

## Tasks

- [x] Add Django settings
- [x] Add ticket model
- [x] Add ticket management UI
- [x] Add UI display/shortcuts for ticket links
- [ ] Test
- [ ] Document where applicable

**Comment by mikkonie** [comment]
unassigned @stolpeo
